core/common/src/main/java/alluxio/collections/LockCache.java

```diff
@@ -174,51 +202,24 @@ public ReentrantReadWriteLock getRawReadWriteLock(K key) {
 
   private ValNode getValNode(K key) {
     Preconditions.checkNotNull(key, "key can not be null");
-    ValNode cacheEntry;
-    while (true) {
-      // repeat until we get a cacheEntry that is not in the process of being removed.
-      cacheEntry = mCache.compute(key, (k, v) -> {
-        if (v != null) {
-          // we want to maintain the invariant that only one thread is removing from the map.
-          // Returning anything other than v would break that invariant.
-          v.mRefCount.incrementAndGet();
-          v.mIsAccessed = true;
```

Comment:
if the above two lines are executed between `mRefCount.compareAndSet(0, INT_MIN)` and `mIterator.erase` in `evictIfOverLimit`, then the returned lock is no longer in the cache, so a further `get` for the same key will get a different lock, which causes locking issues.